---
title: "BIOL 812 Final Assignment - Report"
authors: "Isabella Asselstine, Xinran Li, Aakanx Panchal, Dan Quesnelle"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Data source: <https://www.ncbi.nlm.nih.gov/gene?term=nifH%5BGene%20Name%5D>

GitHub: <https://github.com/dquesnelle/812-Final-Assignment>

Group 1 - Isabella Asselstine, Xinran Li, Aakanx Panchal, Dan Quesnelle

\newpage

## Background

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Methods

NifH gene sequences from 100 different species of bacteria and archaea were collected from the NCBI Gene database. The gene IDs of the 100 gene sequences were compiled into a table along with other identifiers like gene and species name. To isolate the gene IDs from the other components of the table, we ran the file through a script that returns only the gene IDs in numeric form. Itâ€™s in this form that we can use these gene IDs to retrieve the actual sequences. We took advantage of a command-line tool called Entrez Direct to retrieve our sequences. Entrez Direct is part of a larger tool called Entrez that is used for searching and fetching information from NCBI, and is one of the preferred methods for retrieval of genomic, coding and protein sequences (3). We ran the numeric array of gene IDs through an Entrez script that searched NCBI for each  gene sequence, converted them to FASTA format and then outputted them to a file that would end up containing all 100 sequences.


## Results
